packages/model-viewer/src/three-components/SmoothControls.ts

```diff
@@ -823,23 +827,45 @@ export class SmoothControls extends EventDispatcher {
     const {isUserChange} = this;
     this.isUserChange = true;
 
-    switch (event.keyCode) {
-      case KeyCode.PAGE_UP:
+    // Handle modifier key conditions of:
+    // Shift & enablePan == PAN
+    // else == ORBIT/ZOOM
+    relevantKey = (event.shiftKey && this.enablePan) ? this.panKeyCodeHandler(event, isUserChange) : this.orbitZoomKeyCodeHandler(event, isUserChange)
+
+    if (relevantKey) {
+      event.preventDefault();
+    }
+  };
+
+
+
+  /**
+   * Handles the orbit and Zoom key presses
+   * Uses constants for the increment.
+   * @param event The keyboard event for the .key value
+   * @param isUserChange Is this a user initiated event
+   * @returns boolean to indicate if the key event has been handled
+   */
+   private orbitZoomKeyCodeHandler(event: KeyboardEvent, isUserChange: boolean) {
+    // releventKey as the return value is to get the condition back to the calling function as don't have pass by reference arguments in js
```

Comment:
Actually, all arguments in js are passed by reference (except literals like number). How about you move the `event.preventDefault();` into these functions and avoid the return?